fusedrug/data/tokenizer/modulartokenizer/multi_tokenizer.py

```diff
@@ -476,6 +528,75 @@ def save_jsons(self, tokenizers_info: Optional[List] = None) -> None:
 
+    def save(self, path: str) -> None:
+        """Saves all information needed to reconstruct the modular tokenizer to path.
+        After saving, path will contain the following:
+        - json files: modular json files (i.e. that have common special tokens, and that all map to consistent ID space)
+        - config.yaml: a config file that contains the tokenizer_info list with information from self.tokenizer_info, defining the relations between the different tokenizers.
+        ** Since all modular tokenizers are found in the same path, and the path may change when it's passed between users, the path is not included in the config
+        (i.e. all json paths have base dir of './'). The correct path is updated upon calling ModularTokenizer.load()
+
+        Args:
+            path (str): a directory there the modular tokenizer info will be saved.
+        """
+
+        def get_out_path(input_json_path: str, base_path: Optional[str] = None) -> str:
+            """_summary_
+
+            Args:
+                input_json_path (str): _description_
+                base_path (str, optional): _description_. Defaults to None.
+
+            Returns:
+                str: _description_
+            """
+            fname: str = os.path.basename(input_json_path)
+            if base_path is None:
+                return os.path.join("./", fname)
+            return os.path.join(base_path, fname)
+
+        def set_field(tokenizers_info_cfg: List, name: str, key: str, val: Any) -> List:
+            for i, t in enumerate(tokenizers_info_cfg):
+                if t["name"] == name:
+                    t[key] = val
+                    tokenizers_info_cfg[i] = t
+                    return tokenizers_info_cfg
+            raise Exception(f"name {name} not found")
+
+        tokenizers_info_cfg = self.tokenizers_info_raw_cfg
+
+        if not os.path.exists(os.path.dirname(path)):
+            os.makedirs(os.path.dirname(path))
+        for t_type in self.tokenizers_info:
+            tokenizer_inst = self.tokenizers_info[t_type]["tokenizer_inst"]
+            write_out_path = get_out_path(
+                input_json_path=self.tokenizers_info[t_type]["json_path"],
+                base_path=path,
+            )
+            # json paths in the save config are meaningless, since the saves may pass between machines, and the base directory for the save may change.
+            config_out_path = get_out_path(
+                input_json_path=self.tokenizers_info[t_type]["json_path"],
+                base_path=None,
+            )
+            tokenizers_info_cfg = set_field(
+                tokenizers_info_cfg=tokenizers_info_cfg,
+                name=t_type,
+                key="modular_json_path",
+                val=config_out_path,
+            )
+            # Original json path (for the json of the tokenizer used to create the original instance of this ModularTokenizer) is no longer relevant,
```

Comment:
Maybe set to None or "N/A" to better reflect it?
Another option is to copy the original one as well.

Reply:
Done and tested that it works